### ISSUE TITLE
Training iteratively with dynamic resumes

### ISSUE THREAD

**Issue by GabrieldeBlois** [ultralytics/ultralytics#18015]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am fine tuning a yolo11 model with the visDrone dataset. During training, I would like to unfreeze iteratively the weights. Like doing 10 epochs with model freeze=23, 10 epochs with freeze=19, lr=1e-4, and so on.
I tried to launch a first training with the python API and then resume but the epochs argument cannot be changed as soon as I want to use the hub. Somehow the arguments passed to the train function of the model are ignored if the model has been loaded from the hub. Do you have a workaround ?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GabrieldeBlois, thank you for your interest in Ultralytics 🚀! We appreciate your detailed question regarding dynamic iterative training and your use of the Python API with the VisDrone dataset.

If you're encountering behavior where the arguments passed to the `train` function aren’t applied as expected, let’s start by gathering some additional information to help debug 🐛. Could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) that demonstrates the issue? This will allow us to pinpoint the problem effectively.

In the meantime:

## Recommendations
1. Ensure you are using the **latest version** of the `ultralytics` package. Upgrade with:
   ```bash
   pip install -U ultralytics
   ```
2. Verify all your environment dependencies are up to date. Use the dependencies listed in [this file](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml).
   
### Documentation and Resources
- Visit the [Docs](https://docs.ultralytics.com) for detailed usage examples, including approaches for customizing training loops within the [Python API](https://docs.ultralytics.com/usage/python/).
- Review our [Best Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) to ensure your training framework follows recommended practices.

### Collaboration
Join the Ultralytics community to share your insights or discover solutions:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions.
- [Discourse](https://community.ultralytics.com) for detailed question-and-answer threads.
- [Subreddit](https://reddit.com/r/Ultralytics) for broader discussions and community knowledge sharing.

## Environments

To simplify experimentation or debugging, consider running YOLO in one of these reliable and verified environments with all required dependencies pre-installed:
- **Notebooks**: Launch directly with free GPU access:
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Platforms**:
  - [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) Deep Learning VM
  - [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) Deep Learning AMI
- **Docker Images**:
  Run using our prebuilt image: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/).

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a> The current CI status reflects that all automated tests are running successfully across multiple platforms, ensuring core functionality is robust.

This is an automated response to guide you while you wait, but don’t worry—a knowledgeable Ultralytics engineer will assist you soon to clarify your query or provide additional support 😊.

**Comment by GabrieldeBlois** [comment]
Yes, for example if I have made a training through the hub, and I want to continue it with a different configuration, I tried changing manually the config values but it did not work, when I call the train function with `resume=True`, the configuration values of the original training are set up again

**Comment by Y-T-G** [comment]
You can't change the epochs when resuming. You can maybe use callbacks to stop the training when reaching a certain epoch and then resume it.

**Comment by GabrieldeBlois** [comment]
And changing the freeze param ?

**Comment by glenn-jocher** [comment]
The `freeze` parameter can be adjusted between training runs by modifying the model's configuration before starting a new training session, but it cannot be altered dynamically during a single `resume=True` session. To implement iterative unfreezing, train in consecutive runs with updated `freeze` values each time, saving checkpoints at each stage.

**Comment by Y-T-G** [comment]
> And changing the freeze param ?

You can change that while resuming I think.
